packages/notifications/src/redux/reducers/notifications.ts

```diff
@@ -0,0 +1,54 @@
+import { PortalNotificationConfig } from '../../Portal';
+import {
+    ADD_NOTIFICATION,
+    REMOVE_NOTIFICATION,
+    CLEAR_NOTIFICATIONS
+} from '../actions/action-types';
+
+function generateID(type: string): string {
+    return 'cloud-services' + type + '-' + new Date().getTime() + Math.random().toString(36).slice(2);
+}
+
+export interface AddNotificationPayload extends Omit<PortalNotificationConfig, 'id'> {
+    id?: string | number;
+}
+
+const addNotification = (notifications: PortalNotificationConfig[], { payload }: { payload: AddNotificationPayload }): PortalNotificationConfig[] =>
+    [ ...notifications, { id: generateID('notification'), ...payload }];
+
+const removeNotification = (notifications: PortalNotificationConfig[], { payload }: { payload: string | number }) => {
+    const index = notifications.findIndex(({ id }) => id === payload);
+    return [ ...notifications.slice(0, index), ...notifications.slice(index + 1) ];
+};
+
+const clearNotifications = () => [];
+
+export const defaultState: PortalNotificationConfig[] = [];
+
+interface AddNotificationAction {
+    type: typeof ADD_NOTIFICATION,
+    payload: AddNotificationPayload
+}
+
+interface RemoveNotificationAction {
+    type: typeof REMOVE_NOTIFICATION,
+    payload: string | number
+}
+
+export type NotificationsReducerActions = AddNotificationAction | RemoveNotificationAction
+
+export const notificationsReducers = (state = defaultState, action: AddNotificationAction): PortalNotificationConfig[] => {
+    switch (action.type) {
+        case ADD_NOTIFICATION:
+            return addNotification(state, action);
+        case REMOVE_NOTIFICATION:
+            return removeNotification(state, action as unknown as RemoveNotificationAction);
+        case CLEAR_NOTIFICATIONS:
+            return clearNotifications();
+        default:
+            return state;
+    }
```

Comment:
I don't like switches at all, can we use `applyReducerHash`?

Reply:
Again we can do that later. We would first need `applyReducerHash` typings to make it work properly. And honestly, this is such a simple reducer I don't really see any benefit of using the object. Apart from getting rid of the switch startement.

Reply:
Yup, new issue and implement it later.